src/lib/GSENSOR/stk8baxx.cpp

```diff
@@ -0,0 +1,265 @@
+    SRegWriteValue = 0x14; 
+    WriteAccRegister(SRegAddr, SRegWriteValue);
+
+	/* Enable significant motion */
+    SRegAddr       = STK8xxx_REG_SIGMOT2;
+    SRegWriteValue = 0x02; 
+    WriteAccRegister(SRegAddr, SRegWriteValue);
+
+	/* Map significant motion interrupt to INT1 */
+    SRegAddr       = STK8xxx_REG_INTMAP1;
+    SRegWriteValue = 0x01; 
+    WriteAccRegister(SRegAddr, SRegWriteValue);
+
+}
+
+/*
+ * Diable motion register
+ * Must be placed after the software reset
+ *
+ */
+void STK8xxx::STK8xxx_Disable_Motion()
+{
+	unsigned char ARegAddr, ARegWriteValue;
+	
+	/* Disable X Y Z-axis motion (slope) interrupt */
+    ARegAddr       = STK8xxx_REG_INTEN1;
+    ARegWriteValue = 0x00;
+    WriteAccRegister(ARegAddr, ARegWriteValue);
+
+	/* Disable motion */
+    ARegAddr       = STK8xxx_REG_SIGMOT2;
+    ARegWriteValue = 0x00; 
+    WriteAccRegister(ARegAddr, ARegWriteValue);
+}
+
+
+/* Disable Gsensor */
+void STK8xxx::STK8xxx_Suspend_mode()
+{
+	unsigned char RegAddr, RegWriteValue;
+	
+    /* suspend mode enable */
+	RegAddr       = STK8xxx_REG_POWMODE;
+    RegWriteValue = STK_SUSPEND_MODE;
+    WriteAccRegister(RegAddr, RegWriteValue);
+}
+
+bool STK8xxx::STK8xxx_Check_chipid()
+{
+	uint8_t RegAddr = STK_REG_CHIPID;
+    int i = 0, pid_num = (sizeof(stk8xxx_pid_list) / sizeof(stk8xxx_pid_list[0]));
+
+    ReadAccRegister(RegAddr, &chipid_temp);
+    for (i = 0; i < pid_num; i++)
+    {
+        if (chipid_temp == stk8xxx_pid_list[i])
+        {
+        	INFOLN("read stkchip id ok, chip_id = 0x%x", chipid_temp);
+            return true;
+        }
+    }
+	ERRLN("read stkchip id fail!");
+    return false;
+}
+
+/*
+ * Initializes an example function
+ * The initial configuration is in active mode.
+ * STK8xxx is in the gear of ODR=2000Hz, and the range is set to +/-4G
+ *
+ */
+int STK8xxx::STK8xxx_Initialization()
+{
+    unsigned char RegAddr, RegWriteValue;
+
+	if(!STK8xxx_Check_chipid())
+	{
+		return -1;
+    }
+		
+    /* soft-reset */
+	RegAddr       = STK8xxx_REG_SWRST;
+    RegWriteValue = 0xB6;
+    WriteAccRegister(RegAddr, RegWriteValue);
+	delay(50);//unit ms
+		
+    /* set range, resolution */
+    RegAddr       = STK8xxx_REG_RANGESEL;
+    RegWriteValue = STK8xxx_RANGE_4G; // range = +/-4g
+    WriteAccRegister(RegAddr, RegWriteValue);
+
+	/* set power mode */
+	RegAddr 	  = STK8xxx_REG_POWMODE;
+	RegWriteValue = 0x00;	// active mode
+	WriteAccRegister(RegAddr, RegWriteValue);
+	
+	/* set bandwidth */
+    RegAddr       = STK8xxx_REG_BWSEL;
+    RegWriteValue = 0x0F; // bandwidth = 1000Hz
+    WriteAccRegister(RegAddr, RegWriteValue);
+	
+	//STK8xxx_Anymotion_init();
+	//STK8xxx_Sigmotion_init();
+
+    /* set i2c watch dog */
+    RegAddr       = STK8xxx_REG_INTFCFG;
+    RegWriteValue = 0x04; // enable watch dog
+    WriteAccRegister(RegAddr, RegWriteValue);
+
+    /* int config */
+    RegAddr       = STK8xxx_REG_INTCFG1;
+    RegWriteValue = 0x05; // INT1/INT2 push-pull, active high
```

Comment:
Should this be 0x01 @jonesfu ?

![image](https://user-images.githubusercontent.com/14170229/144766716-ad858198-9bf8-48aa-9cb6-3bf2181b32f9.png)

Reply:
Yes, the value should be 0x01.  This code is inherited from old chip.

Reply:
Awesome, thank you.